### PR TITLE
add option to use unix domain socket as Envoy admin endpoint

### PR DESCRIPTION
**Pull request by yxue**
Issue: https://github.com/istio/istio/issues/19684

API change PR: https://github.com/istio/api/pull/1278

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -295,6 +295,10 @@ global:
     # If using stackdriver tracer outside GCP, set env GOOGLE_APPLICATION_CREDENTIALS to the GCP credential file.
     tracer: "zipkin"
 
+    # Specify whether use TCP loopback address as Envoy admin endpoint.
```
Comment (jwendell) [inline install/kubernetes/helm/istio/values.yaml]: Are we still updating helm charts? Those changes should go in operator, no?

**Review by howardjohn — CHANGES_REQUESTED** [review]
it seems like the proper place for this is in mesh config. Is there really a requirement to make it configurable as an annotation and not mesh wide?

File: install/kubernetes/helm/istio/files/injection-template.yaml [hunk 2]
```diff
@@ -180,6 +180,8 @@ containers:
   - --statusPort
   - "{{ annotation .ObjectMeta `status.sidecar.istio.io/port` .Values.global.proxy.statusPort }}"
 {{- end }}
+  - --httpAdminEndpoint
```
Comment (rlenglet) [inline install/kubernetes/helm/istio/files/injection-template.yaml]: This needs to be renamed. Envoy's admin interface always uses HTTP regardless of whether is over TCP or UDS.

File: install/kubernetes/helm/istio/values.yaml [hunk 3]
```diff
@@ -295,6 +295,10 @@ global:
...
+    # If it's set to false, unix domain socket will be used for admin endpoint.
```
Comment (rlenglet) [inline install/kubernetes/helm/istio/values.yaml]: I think this is the wrong logic. This shouldn't be an either/or setting. Envoy should be configured to use UDS unconditionally for the admin interface. The option should be about enabling a TCP proxy to the admin interface in pilot-agent.

**Review by rlenglet — CHANGES_REQUESTED** [review]

**Review by rshriram — CHANGES_REQUESTED** [review]
Repeating what I said in the API PR
I think this PR needs a larger discussion to identify if this is really an issue. It is a question of usability vs paranoia and I would prefer to enable usability first. If you wish, create an annotation to specify the admin endpoint completely. Interested users can specify their own unix domain socket or any other port they wish (it increases the burden for people who are paranoid about this but not for people who are (currently) okay with this).

**Pull request closed without merging**